src/views/Home.jsx

```diff
@@ -1,11 +1,59 @@
+import { generateToken } from '@the-collab-lab/shopping-list-utils';
+import Button from '../components/Button';
 import './Home.css';
+import { useState } from 'react';
+// import { createCollection } from '../api/firebase';
+
+export function Home({ setListToken }) {
+	const [retrievedToken, setRetrievedToken] = useState('');
+
+	/**
+	 * This function generates a new unique Token corresponding to a new shopping list.
+	 */
+	const createNewList = () => {
+		const newToken = generateToken();
+		setListToken(newToken);
+
+		/**
+		 * We were creating a new collection in the database using the newToken:
+		 * createCollection(newToken);
+		 */
+	};
+
+	/**
+	 *  This function ensures that the retrievedToken state always reflects the current value in the input field.
+	 */
+	const handleTokenInput = (e) => {
+		setRetrievedToken(e.target.value);
+	};
+
+	/**
+	 * This function attempts to join a shopping list based on the provided Token.
+	 */
+	const joinExistentList = (e) => {
+		e.preventDefault();
+
+		setListToken(retrievedToken);
+	};
 
-export function Home() {
 	return (
 		<div className="Home">
-			<p>
-				Hello from the home (<code>/</code>) page!
-			</p>
+			<Button label="Create New List" onClick={createNewList} />
+			<p>-or-</p>
+			<p>Join an existing shopping list by entering a three word token.</p>
+
+			<form onSubmit={joinExistentList}>
+				<label htmlFor="token">Share token</label>
+				<br />
+				<input
+					type="text"
+					name="token"
+					onChange={handleTokenInput}
+					value={retrievedToken}
+				/>
+				<br />
+				<Button label="Join an existing list" type="submit" />
```

Comment:
This form is actually part of [#5](https://github.com/the-collab-lab/tcl-63-smart-shopping-list/issues/5) and currently allows me to set my token to an empty string

Reply:
Thank you for all the feedback! Should we add a condition to prevent entering an empty string for now, or is it better to defer this part to issue #5?